gluten-substrait/src/main/scala/org/apache/gluten/execution/BasicPhysicalOperatorTransformer.scala

```diff
@@ -73,9 +77,17 @@ abstract class FilterExecTransformerBase(val cond: Expression, val input: SparkP
       validation: Boolean): RelNode = {
     assert(condExpr != null)
     val args = context.registeredFunction
-    val condExprNode = ExpressionConverter
-      .replaceWithExpressionTransformer(condExpr, attributeSeq = originalInputAttributes)
-      .doTransform(args)
+    val condExprNode = condExpr match {
+      case IncrementMetric(child, metric) =>
```

Comment:
Can we create a class similar as `CHFilterExecTransformer`, such `DeltaFilterExecTransformer` and override `getRelNode` method instead? We should put it under `gluten-delta` folder thus to keep core clean from data lake specific logic and no need add delta package as dependency for `gluten-substrait`.

Reply:
Sure. I will do it after Project supports IncrementMetric